sdk/storage/storage-blob/src/AppendBlobClient.ts

```diff
@@ -312,19 +324,21 @@ export class AppendBlobClient extends BlobClient {
             extractedCreds.accountName!,
             extractedCreds.accountKey
           );
-          urlOrConnectionString = extractedCreds.url + "/" + containerName + "/" + blobName;
+          url = appendToURLPath(
+            appendToURLPath(extractedCreds.url, encodeURIComponent(containerName)),
+            encodeURIComponent(blobName)
+          );
           options.proxy = extractedCreds.proxyUri;
           pipeline = newPipeline(sharedKeyCredential, options);
         } else {
           throw new Error("Account connection string is only supported in Node.js environment");
         }
       } else if (extractedCreds.kind === "SASConnString") {
-        urlOrConnectionString =
-          extractedCreds.url +
-          "/" +
-          containerName +
-          "/" +
-          blobName +
+        url =
+          appendToURLPath(
+            appendToURLPath(extractedCreds.url, encodeURIComponent(containerName)),
```

Comment:
Minor: Encode is fine, while just FYI, containerName has limited charset, and typically doesn't need encode.

Reply:
Yes, containerName doesn't need this since it remains unaffected, still doing this just to be consistent with methods like getContainerClient().